miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/FormPage.java

```diff
@@ -0,0 +1,28 @@
+package uk.ac.bbsrc.tgac.miso.webapp.integrationtest.page;
+
+import java.util.Map;
+
+import org.openqa.selenium.WebDriver;
+
+/**
+ * Page that is mainly composed of a list of fields
+ * 
+ * @param <T> field enum
+ */
+public abstract class FormPage<T> extends HeaderFooterPage {
+
+  public FormPage(WebDriver driver) {
+    super(driver);
+  }
+
+  public abstract boolean isEditable(T field);
```

Comment:
I hadn't done this initially because it forces the enum methods to be public and then they're accessible to the test classes which should not directly access WebElements. It is only test code though, and this change does simplify writing tests, which should probably be the priority, so :+1: